### PR TITLE
Modernize plugin configuration and configure automatic releases

### PR DESCRIPTION
**Pull request by jonesbusy**
This PR refresh/modernize the plugin configuration. Use pom, configure dependabot, configure CD, build against modern Jenkins etc.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1 +1,5 @@
-buildPlugin()
+/*
+ See the documentation for more options:
+ https://github.com/jenkins-infra/pipeline-library/
+*/
+buildPlugin(useContainerAgent: true, platforms: ['linux'], jdkVersions: [11, 17])
```
Comment (jonesbusy) [inline Jenkinsfile:5]: Probably need to keep tests on windows

File: src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java [hunk 2]
```diff
@@ -225,7 +225,6 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
             }
         } catch (IOException ioe) {
             Util.displayIOException(ioe, listener);
-            ioe.printStackTrace(listener.fatalError(hudson.tasks.Messages.CommandInterpreter_CommandFailed()));
```
Comment (jonesbusy) [inline src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java:228]: Cannot access Message anymore. Not sure if there is a replacement or can be removed

File: Jenkinsfile (hunk 1 above)
Comment (MarkEWaite) [reply Jenkinsfile:5]: See the ["Improve a plugin" section on Jenkinsfile](https://www.jenkins.io/doc/developer/tutorial-improve/add-a-jenkinsfile/#create-the-jenkinsfile) for the format that is being used more widely to assure Windows and Linux coverage and JDK 11 and 17.

File: pom.xml [hunk 3]
```diff
@@ -3,168 +3,103 @@
...
-    <pluginRepository>
-      <id>repo.jenkins-ci.org</id>
-      <url>https://repo.jenkins-ci.org/public/</url>
-    </pluginRepository>
-  </pluginRepositories>
-
+  <properties>
+    <changelist>999999-SNAPSHOT</changelist>
+    <gitHubRepo>jenkinsci/ansible-plugin</gitHubRepo>
+    <gitHubRepo>jenkinsci/ansible-plugin</gitHubRepo>
```
Comment (MarkEWaite) [inline pom.xml:25]: Remove the duplicate line?
```suggestion
```

File: src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java (hunk 2 above)
Comment (MarkEWaite) [reply src/main/java/org/jenkinsci/plugins/ansible/AnsibleAdHocCommandBuilder.java:228]: I've inserted the literal text when I encountered this type of mistaken access to messages provided by core.

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Looks good to me, with a few changes for your consideration.

**Comment by MarkEWaite** [comment]
I used my administrator privileges on ci.jenkins.io to run the job with your proposed changes to the Jenkinsfile.

**Comment by jonesbusy** [comment]
> I used my administrator privileges on ci.jenkins.io to run the job with your proposed changes to the Jenkinsfile.

Thanks, I've applied the minor changes. I think we need to run pipeline again with Jenkinsfile changes.

Regards,

File: pom.xml [hunk 4]
```diff
@@ -3,168 +3,101 @@
...
   <description>Ansible support in Jenkins</description>
-  <url>https://github.com/jenkinsci/ansible-plugin</url>
-
-  <licenses>
-    <license>
-      <name>Apache License, Version 2.0</name>
-      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
-      <distribution>repo</distribution>
-    </license>
-  </licenses>
```
Comment (NotMyFault) [inline pom.xml:23]: The license field should remain.

**Review by NotMyFault — CHANGES_REQUESTED** [review]

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**